File: SW-expert-academy/CPP/1219.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
int ans;
vector<int> map[100];
int T,len;

void dfs(int now) {
	if(now == 99) {
		ans = 1;
		return;
	}
	if(map[now].empty())
		return;

for(int i=0;i<map[now].size();i++) 
		dfs(map[now][i]);
}        

int main(){
	for(int test_case = 1; test_case <= 10; test_case++)
	{
    ans = 0;
    int len;
		cin>>T>>len;
		
		for(int i=0;i<len;i++) {
			int a,b;
      cin>>a>>b;
			map[a].push_back(b);
		}
    
		dfs(0);
    
		cout<<"#"<<test_case<<" "<<ans<<endl;	
        for(int i=0;i<100;i++)
           map[i].clear();
    }
}

File: SW-expert-academy/JAVA/1219.java
import java.io.FileInputStream;
import java.util.Scanner;

public class Solution {
	public static int ans;
	public static boolean map[][];
	
	public static boolean empty(boolean[] arr) {
		for(int i=0;i<arr.length;i++)
			if(arr[i])
				return false;
		return true;
	}
	
	public static void dfs(int now) {
		if(now == 99) {
			ans = 1;
			return;
		}
		if(empty(map[now]))
			return;
		
		for(int i=0;i<map[now].length;i++) {
			if(map[now][i])
				dfs(i);
		}
	}
	
	public static void main(String args[]) throws Exception	{		
		Scanner sc = new Scanner(System.in);
		int T;		
		for(int test_case = 1; test_case <= 10; test_case++)
		{
			map = new boolean [100][100];
			ans = 0;
			T=sc.nextInt();
			int len = sc.nextInt();
			
			for(int i=0;i<len;i++) {
				int a = sc.nextInt();
				int b = sc.nextInt();
				map[a][b] = true;
			}
			dfs(0);
			System.out.println("#"+test_case+" "+ans);	
		}
	}	
}
